src/Sarif/Readers/NotificationLevelConverter.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Readers
+{
+    public class NotificationLevelConverter : JsonConverter
```

Comment:
Delete this type. Rename ResultLevelConverter to EnumConverter and let's use for all enums. you will need to update ReadJson to consume the 'objectType' arg. You'll need to update can test type for Enum. We could convert algorithm kind to use this approach (with special handling for names like MD5, which need to lowercase to md5...